versions/1/data_stream/manifest.spec.yml

```diff
@@ -111,6 +111,10 @@ spec:
       type: string
       examples:
       - diagnostics
+    match_all:
```

Comment:
How about `dataset_dynamic` or `dataset_is_pattern` as this only changes whether or not the `dataset` part is a string or a pattern)? 
Releated to the description - we could then change it to something like _indicates whether the dataset name can contain a dynamic part only determined during runtime_.